app/controllers/posts_controller.rb

```diff
@@ -1,39 +1,46 @@
 class PostsController < ApplicationController
-  skip_before_action :verify_authenticity_token
-  load_and_authorize_resource
   def index
-    @user = User.includes(:posts).find(params[:user_id])
+    return unless User.exists?(id: params['user_id'])
+
+    @user = User.find_by(id: params['user_id'])
+    render '../views/layouts/partials/_userposts'
   end
 
   def show
-    @user = User.find(params[:user_id])
-    @post = @user.posts.includes(:comments, :likes).find(params[:id])
-  end
+    return unless Post.exists?(id: params['id'])
 
-  def new
-    @post = Post.new
+    @post = Post.find_by(id: params['id'])
+    if @post.user_id == params['user_id'].to_i
+      render '../views/layouts/partials/_userpostdetails'
+    else
+      render '../views/layouts/partials/_wrongid'
+    end
   end
```

Comment:
- [ ] This rendering way is not appropriate inside the controller. how about doing this:

def create
      @post = @current_user.posts.new(post_params)

      respond_to do |format|
        if @post.save
          format.html { redirect_to @post }
        else
          format.html { render :new }
        end
      end
      
Kindly, refer to the above implementation to work on your controllers.

Reply:
Fixed